core/arch/arm/mm/mobj.c

```diff
@@ -376,11 +377,23 @@ static size_t mobj_reg_shm_get_phys_offs(struct mobj *mobj,
 	return to_mobj_reg_shm(mobj)->page_offset;
 }
 
+static void *mobj_reg_shm_get_va(struct mobj *mobj, size_t offst)
+{
+	struct mobj_reg_shm *mrs = to_mobj_reg_shm(mobj);
+
+	if (!mrs->mm)
+		return NULL;
+
+	return (void *)(tee_mm_get_smem(mrs->mm) + offst + mrs->page_offset);
```

Comment:
`(void *)(vaddr_t)(tee_mm_get_smem(...));`?

Reply:
`tee_mm_get_smem()` returns an `uintptr_t` which is more or less the same thing, I think it's pretty much guaranteed that it will have the same width as a `void *` and not cause a warning.